pkg/template/kurl_context.go

```diff
@@ -0,0 +1,197 @@
+package template
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	kurlclientset "github.com/replicatedhq/kurl/kurlkinds/client/kurlclientset/typed/cluster/v1beta1"
+	kurlv1beta1 "github.com/replicatedhq/kurl/kurlkinds/pkg/apis/cluster/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func GetKurlValues(installerName, nameSpace string) (*kurlv1beta1.Installer) {
+
+	cfg, err := k8sconfig.GetConfig()
+
+	if err != nil {
+		return nil
+	}
+
+	clientset := kurlclientset.NewForConfigOrDie(cfg)
+
+	installers := clientset.Installers(nameSpace)
+
+	retrieved, err := installers.Get(installerName, metav1.GetOptions{})
+
+	if err != nil {
+		fmt.Printf("Could not retrieve installer crd object, does an the installer '%s' exist in the '%s namespace'?\n", installerName, nameSpace)
+		return nil
+	}
+
+	return retrieved
+}
+
+func NewKurlContext(installerName, nameSpace string) (*KurlCtx) {
+	ctx := &KurlCtx{
+		KurlValues: make(map[string]interface{}),
+	}
+
+	retrieved := GetKurlValues(installerName, nameSpace)
+
+	if retrieved != nil {
+		ctx.AddValuesToKurlContext(retrieved)
+	}
+
+	return ctx
+}
+
+func (ctx KurlCtx) AddValuesToKurlContext(retrieved *kurlv1beta1.Installer) {
```

Comment:
This is the generic way the addons and their fields are parsed out using reflection. Basically, it iterates from all addons, then through all their fields, and adds them to a map[string]interface{}. All that is need to be done when changes in kURL are made is to go get the kURL release